qualang_tools/macros/long_wait.py

```diff
@@ -0,0 +1,29 @@
+from typing import Union
+from qm.qua import declare, wait, for_
+
+MAX_WAIT = 2**31 - 1  # units of clock cycles (4ns)
+
+
+def long_wait(wait_time: Union[float, int], *elements: str):
+    """Perform a QUA wait command for longer than the current maximum wait time, 2^31-1.
+
+    This macro unrolls into QUA for-loop of smaller waits if the maximum wait time is exceeded in `wait_time`.
+
+    Note: `wait_time` must strictly be a python literal.
+
+    Args:
+        wait_time (Union[float, int]): Desired wait time in units of clock cycles (4ns).
+        *elements (str): Variable length argument list of elements to wait on.
+    """
+    i = declare(int)
+
+    if not isinstance(wait_time, float | int):
```

Comment:
This can through an error in case of numpy.int32, numpy.int64...
Also, why allowing floats here, since the wait time can only be an integer number of clock cycles? 
Maybe casting it to an integer and raising a warning in case a number with non-zero decimal is provided would be safer, what do you think?

Reply:
Maybe we can compare against the `_Variable` instance from `qm.qua._dsl`.
It's bad practice, but it is being done in other parts (loops) and there's a ticket to expose it properly:
https://quantum-machines.atlassian.net/browse/QUA-243

Reply:
Yes I agree now I am only using `isinstance(variable, (_Variable, _Expression))` to be sure :)

Reply:
Indeed, if it cannot be a QUA variable, it should be fine